File: ambeth/Ambeth.Cache/ambeth/proxy/IValueHolderContainer.cs
using System;
using De.Osthus.Ambeth.Cache;
using De.Osthus.Ambeth.Cache.Model;
using De.Osthus.Ambeth.Merge;
using De.Osthus.Ambeth.Merge.Model;
using De.Osthus.Ambeth.Typeinfo;

namespace De.Osthus.Ambeth.Proxy
{
    public interface IValueHolderContainer : IObjRefContainer
    {
        IObjRelation Get__Self(int relationIndex);

        ICacheIntern __TargetCache { get; set; }

        Object Get__ValueDirect(int relationIndex);

        void Set__ValueDirect(int relationIndex, Object value);
        
        void Set__InitPending(int relationIndex);
        
        void Set__Uninitialized(int relationIndex, IObjRef[] objRefs);
    }
}

File: jambeth/jambeth-cache/src/main/java/de/osthus/ambeth/proxy/IValueHolderContainer.java
package de.osthus.ambeth.proxy;

import de.osthus.ambeth.cache.ICacheIntern;
import de.osthus.ambeth.cache.model.IObjRelation;
import de.osthus.ambeth.merge.model.IObjRef;

public interface IValueHolderContainer extends IObjRefContainer
{
	IObjRelation get__Self(int relationIndex);

	ICacheIntern get__TargetCache();

	void set__TargetCache(ICacheIntern targetCache);

	Object get__ValueDirect(int relationIndex);

	void set__ValueDirect(int relationIndex, Object value);

	void set__InitPending(int relationIndex);

	void set__Uninitialized(int relationIndex, IObjRef[] objRefs);
}
